src/Field.tsx

```diff
@@ -37,38 +57,26 @@ Field.contextTypes = {
   formik: PropTypes.object,
 };
 
-Field.propTypes = {
-  name: PropTypes.string.isRequired,
-  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-};
-
 /**
  * Note: These typings could be more restrictive, but then it would limit the
  * reusability of custom <Field/> components.
- * 
- * @example 
+ *
+ * @example
  * interface MyProps {
  *   ...
  * }
- * 
- * export const MyInput: React.SFC<MyProps & FieldProps> = ({
+ *
+ * export const MyInput: React.SFC<MyProps & FieldComponentProps> = ({
  *   field,
  *   form,
  *   ...props
- * }) => 
+ * }) =>
  *   <div>
  *     <input {...field} {...props}/>
  *     {form.touched[field.name] && form.errors[field.name]}
  *   </div>
  */
-export interface FieldProps {
-  field: {
-    /** Classic React change handler, keyed by input name */
-    onChange: (e: React.ChangeEvent<any>) => void;
-    /** Mark input as touched */
-    onBlur: (e: any) => void;
-    /** Value of the input */
-    value: any;
-  };
+export interface FieldComponentProps {
```

Comment:
I wonder if we should give people the option to type this strictly? 

```tsx
interface FieldComponentProps<Values = any> {
   field: FieldProps,
   form: FormikProps<Values>
}
```

Reply:
Idk if it will be useful but it's not a bad idea to provide to end users this flexibility :)
I will add this.